### ISSUE TITLE
Custom directives not resolved by vue

### ISSUE THREAD

**Issue by BlitZz96** [vue-styleguidist/vue-styleguidist#125]
I'm documenting my component using the `<docs>` tag like this:

```html
<docs>
```jsx
<div class="form-row">
    <kk-textarea label="Label title" sub-label="Label Sub-title" value="Awesome textarea!" />
</div>
```´
</docs>
```

The `<kk-textarea>` uses a custom directive "autoExpand":
```html
<textarea ref="textarea" :value="value" v-autoExpand:now @input="emit"></textarea>
```

I have also included custom css and js via the custom styleguide.config.js

```javascript
require: [
    'babel-polyfill',
    path.join(__dirname, 'public/js/app.js'),
    path.join(__dirname, 'public/css/app.css')
],
```

When I open the styleguide, I get the following error 

> [Vue warn]: Failed to resolve directive: autoExpand

Are custom directives not included?

**Comment by Sound1ab** [comment]
Hi @BlitZz96,

There isn't any option in the config to pass in custom directives that will be registered on the Vue instance. However, it's possible to register these locally in the component.

```
directives: {
  focus: {
    // directive definition
    inserted: function (el) {
      el.focus()
    }
  }
}
```

**Comment by BlitZz96** [comment]
That's a good temporary workaround indeed, @Sound1ab!
I did not think of that. Will definitely do this for now. However, would be great if one could import directives and such aswell as images! 😀

**Comment by rafaesc** [comment]
Yes, first review that the custom directive is installed in `path.join(__dirname, 'public/js/app.js')`

If continues the problem, it's possible that there are several vue versions inside folder node_modules.

vue-styleguidist tries to install vue@^2.4.2, so you have to sure that only exists one vue module.

Works
```
node_modules
      ├── vue
      ├── vue-styleguidist
      │   ├── node_modules
      │   │   ├── ...
```

Error
```
node_modules
      ├── vue
      ├── vue-styleguidist
      │   ├── node_modules
      │   │   ├── vue
```

**Comment by 0m4r** [comment]
I got them working adding them as a (fake) mixin:
in ```styleguide.config.js``` I have added a entry pointing to my ```directive.js```
and inside there
```
import Vue from 'vue'
import myDirective from 'path/to/myDirective'

Vue.use(myDirective)

export default {}
```

not the best solution but it works for me.

**Comment by rafaesc** [comment]
@0m4r @BlitZz96 Ok, I've published the fix under v1.7.9.
Please try it and let me know if it works.

**Comment by 0m4r** [comment]
@rafaesc it would be nice to update the doc as well :)

**Comment by rafaesc** [comment]
@0m4r Yes, that's for sure.

Please, when it will be tested this issue, tell me if I can close it.

**Comment by didagu** [comment]
Does not work with vue 3. Has anyone experienced it with vue 3 yet ?

**Comment by elevatebart** [comment]
@didagu what do you mean "it does not work" ?

Do you mean that 
- the directive does not show up in the docs?
- or the docs are broken and don't open?

[NOTE] The vue-styleguidist library is made for documenting visual components. 
If you need to document your directives, you should create a Markdown file and describe it there.

I hope this helps.
If not, you might have to open a reproduction repo.

**Comment by didagu** [comment]
@elevatebart 
I mean the docs are broken. Here is a sample of my configuration.

`const path = require('path');

module.exports = {
  components: 'src/components/**/[A-Z]*.vue',
  require: [path.join(__dirname, 'styleguide/global.requires.js')],
  template: {
    head: {
      links: [
        {
          rel: 'stylesheet',
          href: 'https://fonts.googleapis.com/css?family=Open+Sans',
        },
      ],
    },
  },
  theme: {
    color: {
      link: '#307b96',
      linkHover: '#57a0ba',
      base: '#2a3539',
      light: '#cccccc',
      lightest: '#e5e5e5',
      border: '#777d81',
      name: '#13ab0e',
      type: '#ef7b90',
      error: '#e42346',
      baseBackground: '#ffffff',
      codeBackground: '#e9f2f5',
      sidebarBackground: '#f0f4f7',
      ribbonBackground: '#fce8ec',
      ribbonText: '#ffffff',
    },
    fontFamily: {
      base: '"Open Sans"',
    },
  },
  styles: {},
};`

**Comment by didagu** [comment]
@elevatebart I mean the docs are broken.
I get an error saying: `TypeError: Cannot read properties of undefined (reading 'deep')`.
That happens when I am using custom directives in my components.
I use to mount them on the vue instance this way:
```
import Vue from 'vue';
import loadable from '../src/directives/loadable';
Vue.directive('loadable', loadable);
```
and include them in the `styleguide.config.js` using: `require: [path.join(__dirname, 'styleguide/global.requires.js')]`
But after upgrading to vue 3, I do this:
```
import { createApp } from 'vue';
import loadable from '../src/directives/loadable';
createApp({}).directive('loadable', loadable)
```
sadly, it does not work as expected. Am I doing something wrong?

**Comment by didagu** [comment]
@elevatebart any update on my question?

**Comment by elevatebart** [comment]
Sorry I missed this.

In vue 3 each vue app it's own little vue instance. If you wanna add directive to it, you cannot do so in a global file.

Fortunately there is an option for customizing the instance used in the examples: https://vue-styleguidist.github.io/Configuration.html#renderrootjsx

**Comment by elevatebart** [comment]
I am not at a computer now. I will try and create a quick demo on Monday.

**Comment by elevatebart** [comment]
Erratum!

After close consideration, the current set of options do not work for global directives. 
I will need to add another option: `enhancePreviewApp`.

Here is how you will use it:

```js
// styleguide.config.js
import { defineConfig } from 'vue-styleguidist'

export default defineConfig({
  enhancePreviewApp: 'styleguide/enhanceApp.js' // path of the file you want to define directives in
})
```

```js
// styleguide/enhanceApp.js
import directive from '@/directives'
import { defineEnhanceApp } from 'vue-styleguidist'

export default defineEnhanceApp((app) => {
  app.directive('directive', directive)
})
```

I will create a separate ticket for it.
